### PR TITLE
Fix TensorRT GridSample inference for LT-DETR

### PR DESCRIPTION
**Pull request by yutong-xiang-97**
## What has changed and why?

This is PR 2 of a three-PR stack and is based on
`yutong-trn-2329-fix-onnx-verification`.

LT-DETR TensorRT engines could build and run successfully while silently producing
incorrect detections or segmentations because TensorRT compiled the deformable-attention
`GridSample` path incorrectly.

For ONNX export, the affected DFINE and RT-DETRv2 attention modules now replace native
`grid_sample` with equivalent gather-based bilinear interpolation built from simpler
operations such as `Pad`, `Gather`, multiplication, and addition. The resulting ONNX
graphs contain no `GridSample` nodes, so no opset-specific TensorRT parser workaround is
needed. Training and ordinary deployment retain the faster native `grid_sample` path.

The ONNX-only conversion is applied through an explicit runtime-checkable export
protocol. DINOv3 ViT-S/16 FP16 TensorRT export remains strongly typed independently of
the `GridSample` fix, preserving the graph's FP32 attention operations and preventing
FP16 overflow. Other model and precision combinations retain the existing weakly typed
path.

## How has it been tested?

Raw ONNX Runtime versus TensorRT results on the T4 with TensorRT 10.13.3.9:

- Object detection:
  - logits max/mean difference: `7.97e-4` / `1.46e-5`
  - boxes max/mean difference: `5.21e-5` / `3.58e-7`
- Instance segmentation:
  - labels: exact match
  - boxes max/mean difference: `1.05e-2` / `1.48e-4`
  - masks max/mean difference: `1.16e-3` / `2.95e-5`
  - scores max/mean difference: `6.87e-6` / `4.29e-7`

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [ ] Yes
- [x] Not needed (the exported model interface is unchanged)

### CODE REVIEW

File: src/lightly_train/_task_models/object_detection_components/dfine_decoder.py [hunk 1]
```diff
@@ -158,6 +158,18 @@ def _reset_parameters(self):
         init.constant_(self.attention_weights.weight, 0)
         init.constant_(self.attention_weights.bias, 0)
 
+    def convert_to_onnx_export(self) -> None:
```
Comment (gabrielfruet) [inline src/lightly_train/_task_models/object_detection_components/dfine_decoder.py:161]: Nit: I would prefer a protocol for that instead of relyinh on `hasattr`.

File: src/lightly_train/_export/tensorrt_helpers.py [hunk 2]
```diff
@@ -263,6 +263,37 @@ def export_tensorrt(
     logger.info("Engine export complete.")
 
 
+def _fix_grid_sample_modes_for_tensorrt(*, onnx_out: PathLike) -> bytes:
```
Comment (gabrielfruet) [inline src/lightly_train/_export/tensorrt_helpers.py]: Since we don't have GridSample anymore anyway at the ONNX, do we need this?

**Review by gabrielfruet — APPROVED** [review]
LGTM
